src/ducks/classifications.js

```diff
@@ -121,10 +120,128 @@ const createClassification = () => {
+      queue via queueClassification()
+    - Then, each item in the queue is individually processed.
+      (The queue is emptied at this point, and a new, empty queue is prepared.)
+      - If the item (Classification) is successfully submitted/saved, that's it.
+        (It's effectively removed from the queue.)
+      - If the item fails, it's reinserted into the new queue, to be processed
+        the next time a new Classification is submitted.
+    - Once ALL the items are processed, we are now ready to prepare our data
+      for the next Subject to be fetched.
+ */
+const saveAllQueuedClassifications = (dispatch, getState) => {
+  const user = getState().login.user;
+  const QUEUE_NAME = (user)
+    ? user.id + '.' + CLASSIFICATIONS_QUEUE_NAME
+    : '_.' + CLASSIFICATIONS_QUEUE_NAME;
+        
+  const queue = JSON.parse(localStorage.getItem(QUEUE_NAME));
+
+  if (queue && queue.length !== 0) {
+    //Prepare 
+    const newQueue = [];
+    localStorage.setItem(QUEUE_NAME, null);  //Empty the queue first
+    
+    //Keep track of items processed so we know when ALL items are processed.
+    let itemsProcessed = 0;
+    let itemsFailed = 0;
+    const itemsToProcess = queue.length;
+    
+    queue.forEach((classificationData) => {
+      //Let's send those Classifications!
+      apiClient.type('classifications').create(classificationData).save()
+      
+      //SUCCESS
+      .then((classificationObject) => {
+        console.info('ducks/classifications.js saveAllQueuedClassifications() success: item ', classificationObject.id);
+
+        try {
+          Split.classificationCreated(classificationObject);
+        } catch (err) { console.error('Split.classificationCreated() error: ', err); }
+
+        //Record locally that the Classification has been seen.
+        const { workflow, subjects } = classificationObject.links;
+        dispatch(addAlreadySeen(workflow, subjects));
+        
+        //OK, we don't need you any more, Classification. Why is this here? It's in PFE.
+        //TODO: check
+        classificationObject.destroy();
+      })
+      
+      //FAILURE
+      .catch((err) => {
+        //Ah, crap.
+        itemsFailed++;
+        
+        console.error('ducks/classifications.js saveAllQueuedClassifications() failure: item ', classificationObject.id);
+        console.error('ducks/classifications.js saveAllQueuedClassifications() error: ', err);
+        Rollbar && Rollbar.error &&
+        Rollbar.error('ducks/classifications.js saveAllQueuedClassifications() error: ', err);
+
+        //Right, why did it fail?
+        switch (err.status) {
+          case 422:  //If the reason for failure is that Panoptes returned a 422, it means the Classification was bad and should be discarded.
+            break;
+
+          default:  //Otherwise, the failed Classification should be re-queued for the next time attempt.
+            newQueue.push(classificationData);
+        }
+      })
+      .finally(() => {
```

Comment:
This Promise function has no IE support.

Reply:
Whaaat? Isn't this polyfilled? Goddammit, let me check on my IE11, then if it effs up... I have a Plan B.

Reply:
Doesn't look like it's currently polyfilled (unless it's somehow in the `es6` polyfill? I didn't test the failure, I'm just assuming based on what's on MDN. If there is a failure, it can easily be polyfilled https://polyfill.io/v2/docs/features/

Reply:
I ran a test using...

```
      .finally(() => {
        //That's one item down.
        itemsProcessed++;

        alert(itemsProcessed + ' of ' + itemsToProcess);
```

...on Win7+IE11 and it looks like it works. Let me know if you still see an issue on your end; if so, we can go with either

A. explicitly putting in that Polyfill for Promise.finally, or

B.
```
const finally = () => {
  ...
};

apiClient.type('classifications').create(classificationData).save()

//SUCCESS
.then(()=> {
  ...
  finally();
})

//FAILURE
.catch((err) => {
  ...
  finally();
});
```

Reply:
I just ran this on my end and all looked well.